src/main/java/com/iota/iri/service/API.java

```diff
@@ -201,14 +225,18 @@ private AbstractResponse process(final String requestString, InetSocketAddress s
             }
 
             if (instance.configuration.string(DefaultConfSettings.REMOTE_LIMIT_API).contains(command) &&
-                    !sourceAddress.getAddress().isLoopbackAddress()) {
+                    !exchange.getSourceAddress().getAddress().isLoopbackAddress()) {
                 return AccessLimitedResponse.create("COMMAND " + command + " is not available on this node");
             }
 
             log.debug("# {} -> Requesting command '{}'", counter.incrementAndGet(), command);
 
             switch (command) {
                 case "storeMessage": {
+                    if(!iotaAPIHeaderDefined(exchange)){
```

Comment:
Does this always appear?
Instead of copy/pasting it can you write it once before the `switch`?

Reply:
I'm going to refactor this. But the idea is not to require the api version header for requests to IXI modeules.

Reply:
The idea is not to require the API version header for calls to IXI modules. Due to the current use of a switch we have 3 options: 
1 - Paste the check for every API call, except call to IXI modules (which is what I've done)
2 - Refactor the current logic to use command pattern
3 - Use a Map structure to map command name to API call (kind of a command pattern)

I can refactor the command structure, but that should be done in another PR.